### ISSUE TITLE
llvm-toolchain-darwin does not support `-s`

### ISSUE THREAD

**Issue by GregBowyer** [llvm/llvm-project#60430]
Hi, I am doing slightly odd things with cross compilation and Bazel.

Currently, Bazel insists on calling `libtool` with `-static -s` for static libraries / archives.

While mostly a drop-in replacement, the current `llvm-libtool-darwin` lacks support for `-s` which in the OSX man page is given as the following:

```
       -s     Produce  the  preferred type of table of contents, which results in faster link editing when linking with the archive.  The order of
	      the table of contents is sorted by symbol name.  The library member name of this type of table of contents is ``__.SYMDEF  SORTED''.
	      This  type  of  table  of contents can only be produced when the library does not have multiple members that define the same symbol.
	      This is the default.
```

If this is what `llvm-libtool-darwin` does naturally (since it is the default), could this be a compatibility NOP flag?

**Comment by keith** [comment]
I agree that we should fix this, but on the bazel side I don't see where `-s` is coming from either, maybe you're on an older version?

**Comment by keith** [comment]
Looks like you might be getting it from here https://github.com/bazelbuild/bazel/pull/17489 which I think we can remove. If so that likely means you're using a custom C++ toolchain (or maybe [bazel-toolchain](https://github.com/grailbio/bazel-toolchain)) and it's getting this flag. A short term workaround would be to define this feature in the crosstool and leave this flag out (the default crosstool does not pass it or inherit this behavior).

As for the LLVM side it looks to me like the current behavior of llvm-libtool-darwin is actually the opposite and it always uses the `-a` behavior, never producing a `SYMDEF SORTED` file https://github.com/llvm/llvm-project/blob/639b7865a67f886ca9d7b994d577236a6957734c/llvm/lib/Object/ArchiveWriter.cpp#L385-L399

Seems to me like there are a few options:

1. Add true support for `-s` vs `-a`, probably ideal, but the most work
2. Add argument handling for `-s` and warn that it's not supported, maybe a bit noisy for your case
3. Add argument handling for `-s` and silently ignore it, would solve your case but if someone really cares about the `-s` vs `-a` difference, it might be surprising.

**Comment by GregBowyer** [comment]
I am using a heavily derived toolchains from grailbio (or once upon a time from grailbio).

I did work around it on my end, not sure what it takes to implement `-s` correctly. Furthermore, I would agree with you that the failing silently would be a surprise to most people.
